subcommand/consul-sidecar/command.go

```diff
@@ -80,29 +95,66 @@ func (c *Command) Run(args []string) int {
 		c.UI.Error(err.Error())
 		return 1
 	}
+	c.logger = logger
 
 	// Log initial configuration
-	logger.Info("Command configuration", "service-config", c.flagServiceConfig,
+	c.logger.Info("Command configuration", "service-config", c.flagServiceConfig,
 		"consul-binary", c.flagConsulBinary,
 		"sync-period", c.flagSyncPeriod,
-		"log-level", c.flagLogLevel)
+		"log-level", c.flagLogLevel,
+		"enable-metrics-merging", c.flagEnableMetricsMerging,
+		"merged-metrics-port", c.flagMergedMetricsPort,
+		"service-metrics-port", c.flagServiceMetricsPort,
+		"service-metrics-path", c.flagServiceMetricsPath,
+	)
 
 	c.consulCommand = []string{"services", "register"}
 	c.consulCommand = append(c.consulCommand, c.parseConsulFlags()...)
 	c.consulCommand = append(c.consulCommand, c.flagServiceConfig)
 
-	// ctx that we pass in to the main work loop, signal handling is handled in another thread
+	// signalCtx that we pass in to the main work loop, signal handling is handled in another thread
 	// due to the length of time it can take for the cmd to complete causing synchronization issues
 	// on shutdown. Also passing a context in so that it can interrupt the cmd and exit cleanly.
-	ctx, cancelFunc := context.WithCancel(context.Background())
+	signalCtx, cancelFunc := context.WithCancel(context.Background())
 	go func() {
 		select {
 		case sig := <-c.sigCh:
-			logger.Info(fmt.Sprintf("%s received, shutting down", sig))
+			c.logger.Info(fmt.Sprintf("%s received, shutting down", sig))
 			cancelFunc()
 			return
 		}
 	}()
+
```

Comment:
TODO: needs more logging, it's hard to tell if this is starting up without errors and if it's doing ok scraping metrics